provisioning/service/src/Config/EnrollmentGroup.cs

```diff
@@ -0,0 +1,313 @@
+    ///    The minimum information required by the provisioning service is the {@link #enrollmentGroupId} and the
+    ///    {@link #attestation}.
+    ///
+    /// To provision a device using EnrollmentGroup, it must contain a X509 chip with a signingCertificate for the
+    ///    {@link X509Attestation} mechanism.
+    ///
+    /// The content of this class will be serialized in a JSON format and sent as a body of the rest API to the
+    ///    provisioning service.
+    ///    
+    /// The content of this class can be filled by a JSON, received from the provisioning service, as result of a
+    ///    EnrollmentGroup operation like create, update, or query EnrollmentGroup.
+    /// </remarks>
+    /// <example>
+    /// When serialized, an EnrollmentGroup will look like the following example:
+    /// <code>
+    /// {
+    ///    "enrollmentGroupId":"validEnrollmentGroupId",
+    ///    "attestation":{
+    ///        "type":"x509",
+    ///        "signingCertificates":{
+    ///            "primary":{
+    ///                "certificate":"[valid certificate]"
+    ///            }
+    ///        }
+    ///    },
+    ///    "iotHubHostName":"ContosoIoTHub.azure-devices.net",
+    ///    "provisioningStatus":"enabled"
+    /// }
+    /// </code>
+    ///
+    /// The following JSON is a sample of the EnrollmentGroup response, received from the provisioning service.
+    /// <code>
+    /// {
+    ///    "enrollmentGroupId":"validEnrollmentGroupId",
+    ///    "attestation":{
+    ///        "type":"x509",
+    ///        "signingCertificates":{
+    ///            "primary":{
+    ///                "certificate":"[valid certificate]",
+    ///                "info": {
+    ///                    "subjectName": "CN=ROOT_00000000-0000-0000-0000-000000000000, OU=Azure IoT, O=MSFT, C=US",
+    ///                    "sha1Thumbprint": "0000000000000000000000000000000000",
+    ///                    "sha256Thumbprint": "validEnrollmentGroupId",
+    ///                    "issuerName": "CN=ROOT_00000000-0000-0000-0000-000000000000, OU=Azure IoT, O=MSFT, C=US",
+    ///                    "notBeforeUtc": "2017-11-14T12:34:18Z",
+    ///                    "notAfterUtc": "2017-11-20T12:34:18Z",
+    ///                    "serialNumber": "000000000000000000",
+    ///                    "version": 3
+    ///                }
+    ///            }
+    ///        }
+    ///    },
+    ///    "iotHubHostName":"ContosoIoTHub.azure-devices.net",
+    ///    "provisioningStatus":"enabled",
+    ///    "createdDateTimeUtc": "2017-09-28T16:29:42.3447817Z",
+    ///    "lastUpdatedDateTimeUtc": "2017-09-28T16:29:42.3447817Z",
+    ///    "etag": "\"00000000-0000-0000-0000-00000000000\""
+    /// }
+    /// </code>
+    /// </example>
+    /// <seealso cref="https://docs.microsoft.com/en-us/rest/api/iot-dps/deviceenrollmentgroup">Device Enrollment Group</seealso>
+    class EnrollmentGroup : IETagHolder
```

Comment:
Missing mandatory `internal` access modifier.

Reply:
Actually: shouldn't this be 'public'? (Default is `internal`.)

Reply:
Public, done